be/src/storage/rowset/beta_rowset_writer.cpp

```diff
@@ -231,6 +239,7 @@ StatusOr<std::unique_ptr<SegmentWriter>> HorizontalBetaRowsetWriter::_create_seg
     writer_options.storage_format_version = _context.storage_format_version;
     const auto* schema = _rowset_schema != nullptr ? _rowset_schema.get() : _context.tablet_schema;
     writer_options.global_dicts = _context.global_dicts != nullptr ? _context.global_dicts : nullptr;
+    writer_options.column_indexes = _context.column_indexes;
```

Comment:
For column_indexes is a vector, and writer_options keep the same for one rowset writer, So I think write_options can be optimized to create once to avoid copy